src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    const { type } = action;
+
+    switch (type) {
+      case 'addProperties':
+        currentState = {
+          ...currentState, ...action.extraData,
+        };
+        result.push({ ...currentState });
+        break;
+
+      case 'removeProperties':
```

Comment:
for types you can create an object like 

```
const ACTION_TYPES = {
    add: 'addProperties',
    remove: 'removeProperties',
    clear: 'clear',
  };
```
  
  and use types as keys of object in switch case